### PR TITLE
Update websockets library to work with Py3.9+

### PR DESCRIPTION
**Pull request by johnsca**
Using websockets < 8.0 on Python >= 3.9 fails with a TypeError in the websockets library. However, version 8 of websockets drops support for older versions of Python that we still need to support, so this uses conditional dependencies to support both.

### CODE REVIEW

**Comment by SimonRichardson** [comment]
Does this also help with #461 (not checked the differences supporting 3 9+)?

**Comment by johnsca** [comment]
@SimonRichardson Alas, no.  The loop arguments will still need to be explicitly removed before 3.10

File: tox.ini [hunk 1]
```diff
@@ -27,8 +27,9 @@ deps =
     pytest-asyncio
     pytest-xdist
     Twine
+    websockets
     # use fork to pick up fix for https://github.com/aaugustin/websockets/pull/528
-    git+https://github.com/johnsca/websockets@bug/client-redirects#egg=websockets
+    git+https://github.com/johnsca/websockets@bug/client-redirects#egg=websockets ; python_version<'3.9'
 
 [testenv:py{3,35,36,37,38}]
```
Comment (SimonRichardson) [inline tox.ini:34]: @johnsca can we also add 39 to this list as well?

**Comment by SimonRichardson** [comment]
!!build!!

Try again...

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
!!build!!

Sorry about the noise, just setting our jenkins CI python 3.9 ready.

**Comment by SimonRichardson** [comment]
@johnsca can you rebase your branch so we can land this.

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
@johnsca The reason the `github-check-merge-juju-python-libjuju` test is failing is that it can't find pylxd and we need this one to pass so I can get the merge bot to land it.

Not entirely sure this isn't being picked up for 3.9, but is for other versions.

```
09:22:21 ==================================== ERRORS ====================================
09:22:21 _______________ ERROR collecting tests/integration/test_model.py _______________
09:22:21 ImportError while importing test module '/home/ubuntu/go/src/github.com/juju/python-libjuju/tests/integration/test_model.py'.
09:22:21 Hint: make sure your test modules/packages have valid Python names.
09:22:21 Traceback:
09:22:21 /usr/lib/python3.9/importlib/__init__.py:127: in import_module
09:22:21     return _bootstrap._gcd_import(name[level:], package, level)
09:22:21 tests/integration/test_model.py:14: in <module>
09:22:21     import pylxd
09:22:21 E   ModuleNotFoundError: No module named 'pylxd'
```

File: tox.ini (hunk 1 above)
Comment (johnsca) [reply tox.ini]: Ah, this was the reason for the pylxd failure below.  This version-specific section actually isn't necessary; it's redundant with the default testenv, so I just folded them together.

**Review by SimonRichardson — APPROVED** [review]

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**